### PR TITLE
[datareposink] Add json property

### PR DESCRIPTION
**Pull request by songgot**
- Add setting JSON file path to write the meta information of a sample data
 - Write gstreamer caps, sample size, and number of total samples
 - In the case of flexible tensors, write the offset of each sample except the sample size

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

### CODE REVIEW

File: gst/datarepo/gstdatareposink.c [hunk 1]
```diff
@@ -524,13 +574,19 @@ gst_data_repo_sink_open_file (GstDataRepoSink * sink)
...
   filename = g_strdup (sink->filename);
 
+  /* for image, g_file_set_contents() is used in the write function */
+  if (sink->media_type == _NNS_IMAGE) {
+    return TRUE;
+  }
+
   GST_INFO_OBJECT (sink, "opening file %s", filename);
 
   /** How about support file mode property ?
```
Comment (again4you) [inline gst/datarepo/gstdatareposink.c:592]: Unnecessary comment?

File: gst/datarepo/gstdatareposink.c [hunk 2]
```diff
@@ -578,30 +634,156 @@ gst_data_repo_sink_open_file (GstDataRepoSink * sink)
 }
 
 /**
- * @brief Start datareposink
+ * @brief Steop datareposink
```
Comment (again4you) [inline gst/datarepo/gstdatareposink.c:637]: typo

Comment (songgot) [reply gst/datarepo/gstdatareposink.c:637]: Thank you.

File: gst/datarepo/gstdatareposink.c (hunk 1 above)
Comment (songgot) [reply gst/datarepo/gstdatareposink.c:592]: No, Thank you.

**Review by niley7464 — APPROVED** [review]
LGTM :)

**Comment by jaeyun-jung** [comment]
plz do rebase, if needed.

**Review by jaeyun-jung — APPROVED** [review]

**Review by again4you — APPROVED** [review]
LGTM!

**Pull request merged**